src/satosa/internal_data.py

```diff
@@ -86,7 +87,7 @@ def hash_id(salt, user_id, requester, state):
             user_id = "{req}{time}{id}".format(req=requester, time=timestamp, id=user_id)
         elif hash_type == UserIdHashType.persistent or hash_type == UserIdHashType.pairwise:
             user_id = "{req}{id}".format(req=requester, id=user_id)
-        elif hash_type == UserIdHashType.public:
+        elif hash_type in (UserIdHashType.public, UserIdHashType.public_email):
```

Comment:
If we're going to use this format (`x in y`) for multiple comparisons, we should do the same (in one go) a bit above, on [line 88](https://github.com/SUNET/SATOSA/pull/137/files#diff-bc07108acca2b02cc3d65ee9df483bf1R88) to keep things consistent.